tests/unit/plugins/filter/test_ipsubnet.py

```diff
@@ -94,3 +94,73 @@ def test_ipvsubnet_filter_rank_address_in_subnet3(self):
         args = ["", "192.168.144.16/30", "192.168.144.0/24"]
         result = _ipsubnet(*args)
         self.assertEqual(result, "5")
+
+    def test_ipvsubnet_get_subnet(self):
+        test_cases = [
+            [
+                ["", "2600:1f1c:1b3:8f00::/56", 64, 0],
+                "2600:1f1c:1b3:8f00::/64",
+            ],
+            [
+                ["", "2600:1f1c:1b3:8f00::/56", "64", "1"],
+                "2600:1f1c:1b3:8f01::/64",
+            ],
+            [
+                ["", "2600:1f1c:1b3:8f00::/56", "64", "2"],
+                "2600:1f1c:1b3:8f02::/64",
+            ],
+            [
+                ["", "2600:1f1c:1b3:8f00::/56", "64", "-2"],
+                "2600:1f1c:1b3:8ffe::/64",
+            ],
+            [
+                ["", "2600:1f1c:1b3:8f00::/56", "64", "-1"],
+                "2600:1f1c:1b3:8fff::/64",
+            ],
+            [
+                ["", "2600:1f1c:1b3:8f00::/56", "65", "0"],
+                "2600:1f1c:1b3:8f00::/65",
+            ],
+            [
+                ["", "2600:1f1c:1b3:8f00::/56", "65", "1"],
+                "2600:1f1c:1b3:8f00:8000::/65",
+            ],
+            [
+                ["", "2600:1f1c:1b3:8f00::/56", "65", "2"],
+                "2600:1f1c:1b3:8f01::/65",
+            ],
+            [
+                ["", "2600:1f1c:1b3:8f00::/56", "65", "-2"],
+                "2600:1f1c:1b3:8fff::/65",
+            ],
+            [
+                ["", "2600:1f1c:1b3:8f00::/56", "65", "-1"],
+                "2600:1f1c:1b3:8fff:8000::/65",
+            ],
+            [
+                ["", "2600:1f1c:1b3:8f00::/56", "64", "257"],
+                False,
+            ],  # subnet id too big for /64
+            [
+                ["", "2600:1f1c:1b3:8f00:::/56", "64", "0"],
+                False,
+            ],  # Too many colons
+            [
+                ["", "2600:1f1c:1b3:8f00::/56", "129", "0"],
+                False,
+            ],  # Must be /128 or less
+            [
+                ["", "2600:1f1c:1b3:8f00::/56", "120", "0"],
```

Comment:
This unit test with a `/120` prefix length causes the `ipsubnet` filter to use 100% CPU utilization.

Reply:
can you open another issue for this and in this prs only doc changes and other uts?

Reply:
You mean another issue in addition to #132 ?